guides/21_Background_Filter_VideoFX_Processor.md

```diff
@@ -75,38 +75,34 @@ The following table lists the browsers and version that support background filte
-+ `worker-src:` add `blob:` to load worker JavaScript across origins\.
-+ Safari only: `child-src:` add `blob:` to load worker JavaScript across origins\.
-
-If you omit any of these entries, or if you use HTTP headers and `http-equiv` meta tags to specify a policy and inadvertently exclude any of these by intersection, then a background filter will not be able to initialize\. The filter appears to be unsupported, or it creates a no\-op video frame processor\. You will see errors in your browser console such as:
-
+# Using a Content Security Policy<a id="content-security"></a>
+Modern web applications use Content Security Policy to protect users from certain classes of attacks. An application that utilizes the `VideoFxProcessor` will need to include the following in policy directives to allow the Amazon Chime SDK access to resources it needs at runtime:
+## Required CSP Directives ##
+### Script Policy ###
+To function, the `VideoFxProcessor` class must load JavaScript classes at runtime from an Amazon\-owned CDN\. These classes implement post\-processing for video using WebGL2\. To grant an application access to fetching and running these classes, you must include the following directives:
 ```
-Refused to connect to
-'https://static.sdkassets.chime.aws/bgblur/workers/worker.js…'
-because it violates the document's content security policy.
+script-src       'self' blob: https://*.sdkassets.chime.aws
+script-src-elem  'self' blob: https://*.sdkassets.chime.aws
 ```
-
-The following example shows a typical content security policy\. It includes the tags needed to access the resources needed to run the filters\. You can adapt this example as needed\. 
-
+Please note that `script-src` and `script-src-elem` directives are both required for full support on Safari and Firefox browsers\.
+### Worker Policy ###
+The `VideoFxProcessor` loads JavaScript classes as a blob to run a web worker thread, which processes video using Machine Learning (ML) models\. To grant an application access to fetching and using this worker, include:
+```
+worker-src       'self' blob: https://*.sdkassets.chime.aws
+```
+### WebAssembly Policy ###
+The `VideoFxProcessor` loads a WebAssembly (Wasm) module from the same Amazon-owned CDN\. In Chrome 95 and later, compiled Wasm modules cannot be passed across multiple module boundaries\. To allow fetching and instantiating these modules, please also include `'wasm-unsafe-eval'` to the `script-src` directive\. 
+
+One can read further on the Content Security Policy’s documentation for WebAssembly [here](https://github.com/WebAssembly/content-security-policy/blob/main/proposals/CSP.md)\.  Ideally the `wasm-eval` tag would be better used, however it yet to gain official standardization. Correspondence on the directive can be found [here](https://bugs.chromium.org/p/v8/issues/detail?id=7041)\.
```

Comment:
So we do not need both `wasm-eval ` and `wasm-unsafe-eval` as before?

Reply:
Correct. `wasm-eval` is not yet standardized and will often times produce an error along the lines of:
 `The source list for Content Security Policy directive <directive-x> contains an invalid source: ''wasm-eval''. It will be ignored.`